dispatch/base.h

```diff
@@ -254,4 +254,16 @@
 
 typedef void (*dispatch_function_t)(void *_Nullable);
 
+#ifdef __SIZEOF_POINTER__
+#define DISPATCH_PTR_SIZE __SIZEOF_POINTER__
+#elif defined(_WIN64)
+#define DISPATCH_PTR_SIZE 8
+#elif defined(_WIN32)
+#define DISPATCH_PTR_SIZE 4
+#elif defined(_MSC_VER)
+#error "could not determine pointer size as a constant int for MSVC"
+#else
+#define DISPATCH_PTR_SIZE sizeof(void *)
```

Comment:
you can't do that, it prevents doing comparisons with the preprocessor which is the whole point.

do
```
#elif defined(__LP64__) || defined(__LLP64__)
#define ... 8
#elif defined(__ILP32__)
#define ... 4
#else
#error ...
#endif // __SIZEOF_POINTER__
````